### ISSUE TITLE
Slice time axis after convolution

### ISSUE THREAD

**Issue by vieting** [rwth-i6/pytorch-to-returnn#119]
I encountered an issue for a combination of padding, convolution and slicing. Specifically, the time axis is padded before convolution and sliced afterwards to ensure that it has the same length as before.

A simple test case to reproduce this is below. I'll open a PR with this test case and provide stack trace and log there.

```
def test_pad_conv():
  n_batch, n_time, n_feat = 3, 7, 5
  conv_pos = 2

  def model_func(wrapped_import, inputs: torch.Tensor):
    if typing.TYPE_CHECKING or not wrapped_import:
      import torch
    else:
      torch = wrapped_import("torch")

    mod_pad = torch.nn.ConstantPad1d(conv_pos // 2, 0)
    inputs = mod_pad(inputs)
    mod_conv = torch.nn.Conv1d(n_feat, n_feat, kernel_size=conv_pos)
    inputs = mod_conv(inputs)
    inputs = inputs[:, :, :-1]
    return inputs

  x = numpy.ones((n_batch, n_feat, n_time)).astype("float32")
  verify_torch_and_convert_to_returnn(model_func, inputs=x)
```

**Comment by vieting** [comment]
> Is this a RETURNN issue? Can you reproduce it in a pure net dict? Then please open a RETURNN issue on this.

Yes, it is and I opened an issue with a pure RETURNN test case here: https://github.com/rwth-i6/returnn/issues/1017

**Comment by vieting** [comment]
Fixed via https://github.com/rwth-i6/returnn/pull/1018, see https://github.com/rwth-i6/returnn/issues/1017.
